lib/common.js

```diff
@@ -365,27 +365,20 @@ const addHeaderLine = function(headers, name, value) {
  * @fieldName {String} field name - string with the case-insensitive field name
  */
 const deleteHeadersField = function(headers, fieldNameToDelete) {
-  if (!_.isObject(headers) || !_.isString(fieldNameToDelete)) {
-    // TODO-coverage: For `_.isObject(headers)`, add a test to cover the
-    // missing condition, or remove if not reachable. For
-    // `_.isString(fieldNameToDelete)`, throw an error and add a test covering
-    // this case.
-    return
+  if (!_.isPlainObject(headers)) {
+    throw Error('headers must be an object')
+  }
+
+  if (!_.isString(fieldNameToDelete)) {
```

Comment:
Could you perform this type check using plain JS?

Reply:
I can. 
I'm still not sure when we want to be using Lodash or not.
For this particular case, because `_.isString` is technically more robust than `typeof var === 'string'`, I thought it would be a good candidate to keep using Lodash.

Reply:
I'm fine with that for now. Ultimately I'd lean toward removing it entirely, though type checking is an area where it's not clear to me whether we need something more robust than `typeof`.

Reply:
The issue with string is
```js
var stringValue = "This is a String";
var stringObject = new String("This is a String Object");
typeof stringValue === "string"
typeof stringObject === "object"
```
The "proper" native js check would be
```js
typeof value === 'string' || value instanceof String
```